src/components/self-referential-canonical-tag/index.tsx

```diff
@@ -0,0 +1,18 @@
+import Head from 'next/head'
+import { useRouter } from 'next/router'
+
+const useFullURL = (base: string = 'https://developer.hashicorp.com') => {
+	const { asPath } = useRouter()
+	return new URL(asPath, base).toString()
+}
+
+const SelfReferentialCanonicalTag = () => {
+	const canonicalURL = useFullURL()
+	return (
+		<Head key={canonicalURL}>
+			<link rel="canonical" href={canonicalURL} />
```

Comment:
I think we can add a `key` prop here to ensure canonicals rendered in page components take priority. We might also have to add the same key to existing canonical tags rendered.

```suggestion
			<link rel="canonical" href={canonicalURL} key="canonical" />
```

ref: https://nextjs.org/docs/api-reference/next/head

Reply:
Added: https://github.com/hashicorp/dev-portal/pull/969/commits/fef495a8670901cd6b0790b291fcf1b469b49b90